sway-lib-std/src/convert.sw

```diff
@@ -1,8 +1,17 @@
 //! Traits for conversions between types.
 library convert;
 
+use ::option::Option;
+
 /// Used to do value-to-value conversions while consuming the input value.
 pub trait From<T> {
     fn from(b: T) -> Self;
     fn into(self) -> T;
 }
+
+// TODO: return a Result when https://github.com/FuelLabs/sway/issues/610 is resolved
+/// Used to attempt to do value-to-value conversions while consuming the input value.
```

Comment:
Question: the test indicates that the length and capacity remain despite returning `None`, should this be updated as the value is not consumed or am I misunderstanding?

Reply:
cc @mohammadfawaz What's your take on this? should the docstrings for `From` & `TryFrom` be updated to NOT say they consume the input value, or should we be enforcing that the value is in fact consumed (by clearing the value or something)?

Reply:
Yeah I see that Rust uses "consume" but that's probably due to the borrow checker. For us, it's probably better to avoid using "consume" and just say "Used to attempt to do value-to-value conversions".